library/src/main/java/org/xmtp/android/library/Contacts.kt

```diff
@@ -125,80 +163,95 @@ class ConsentList(
-                                it.setDenyGroup(
-                                    PrivatePreferencesAction.DenyGroup.newBuilder()
-                                        .addGroupIds(entry.value),
-                                )
-
-                            ConsentState.UNKNOWN -> it.clearMessageType()
+                        EntryType.GROUP_ID -> {
+                            when (entry.consentType) {
+                                ConsentState.ALLOWED ->
+                                    it.setAllowGroup(
+                                        PrivatePreferencesAction.AllowGroup.newBuilder()
+                                            .addGroupIds(entry.value),
+                                    )
+
+                                ConsentState.DENIED ->
+                                    it.setDenyGroup(
+                                        PrivatePreferencesAction.DenyGroup.newBuilder()
+                                            .addGroupIds(entry.value),
+                                    )
+
+                                ConsentState.UNKNOWN -> it.clearMessageType()
+                            }
                         }
-                    }
 
-                    ConsentListEntry.EntryType.INBOX_ID -> {
-                        when (entry.consentType) {
-                            ConsentState.ALLOWED ->
-                                it.setAllowInboxId(
-                                    PrivatePreferencesAction.AllowInboxId.newBuilder()
-                                        .addInboxIds(entry.value),
-                                )
-
-                            ConsentState.DENIED ->
-                                it.setDenyInboxId(
-                                    PrivatePreferencesAction.DenyInboxId.newBuilder()
-                                        .addInboxIds(entry.value),
-                                )
-
-                            ConsentState.UNKNOWN -> it.clearMessageType()
+                        EntryType.INBOX_ID -> {
+                            when (entry.consentType) {
+                                ConsentState.ALLOWED ->
+                                    it.setAllowInboxId(
+                                        PrivatePreferencesAction.AllowInboxId.newBuilder()
+                                            .addInboxIds(entry.value),
+                                    )
+
+                                ConsentState.DENIED ->
+                                    it.setDenyInboxId(
+                                        PrivatePreferencesAction.DenyInboxId.newBuilder()
+                                            .addInboxIds(entry.value),
+                                    )
+
+                                ConsentState.UNKNOWN -> it.clearMessageType()
+                            }
                         }
                     }
                 }
-            }
-        }.build()
+            }.build()
```

Comment:
cool trick, in kotlin you can use `to` infix notation to create `Pair`s to make this more concise:

```kotlin
when (entry.entryType to entry.consentType) {
                        EntryType.ADDRESS to ConsentState.ALLOWED -> setAllowAddress(
                            PrivatePreferencesAction.AllowAddress.newBuilder()
                                .addWalletAddresses(entry.value)
                        )

                        EntryType.ADDRESS to ConsentState.DENIED -> setDenyAddress(
                            PrivatePreferencesAction.DenyAddress.newBuilder()
                                .addWalletAddresses(entry.value)
                        )

                        EntryType.GROUP_ID to ConsentState.ALLOWED -> setAllowGroup(
                            PrivatePreferencesAction.AllowGroup.newBuilder()
                                .addGroupIds(entry.value)
                        )

                        EntryType.GROUP_ID to ConsentState.DENIED -> setDenyGroup(
                            PrivatePreferencesAction.DenyGroup.newBuilder().addGroupIds(entry.value)
                        )

                        EntryType.INBOX_ID to ConsentState.ALLOWED -> setAllowInboxId(
                            PrivatePreferencesAction.AllowInboxId.newBuilder()
                                .addInboxIds(entry.value)
                        )

                        EntryType.INBOX_ID to ConsentState.DENIED -> setDenyInboxId(
                            PrivatePreferencesAction.DenyInboxId.newBuilder()
                                .addInboxIds(entry.value)
                        )

                        else -> clearMessageType()
                    }
                }
            }.build()
```

not that common though so up to you